website/docs/reference/datasources/hubspot/README.md

```diff
@@ -0,0 +1,152 @@
+
+# HubSpot
+
+
+HubSpot is a **cloud-based CRM** designed to help align sales and marketing teams, promote sales enablement, increase ROI, and improve your inbound marketing strategy to produce more qualified leads. In simple words, it is a software platform created to assist your business with marketing and sales.
+
+:::info
+To integrate Hubspot with Appsmith, you should understand the [basics of connecting to databases](core-concepts/connecting-to-data-sources/) on Appsmith.
+:::
+
+Appsmith provides an easy way to integrate with HubSpot, and enables you to perform different operations using a custom UI built with minimal configurations.
+
+
+# Create HubSpot Datasource
+
+To add a HubSpot API, navigate to **Explorer** >> Click on the **+** icon (next to Datasources) >> Select **HubSpot API** under APIs.
+
+<figure>
+ <object data="https://www.youtube.com/embed/h2Jp8PenbTM" width='750px' height='400px'></object>
+ <figcaption align="center"><i>Create Hubspot Datasource</i></figcaption>
+</figure>
+
+
+
+## Connection Settings
+
+Configure the Hubspot API as illustrated below:
+
+![](/img/hubspot-connection.png)
+
+
+
+### Authentication Type
+Appsmith allows you to select from the available authentication types to integrate with an Hubspot database:
+
+#### **Bearer Token** 
+
+Bearer Tokens are the predominant access token used with OAuth 2.0.
+This is a single string that is the authentication of the API request, sent in an HTTP “Authorization” header. You can authenticate to the API by providing your API key in the HTTP authorization bearer token header. 
+
+Let's see how you can fetch your Bearer Token:
+
+
+
+<figure>
+ <object data="https://www.youtube.com/embed/q1DkGLJ6JFU" width='750px' height='400px'></object>
+ <figcaption align="center"><i>Fetching Access Token</i></figcaption>
+</figure>
+
+
+* Open your [**HubSpot Developer Account**](https://developers.hubspot.com/docs/api/overview).
+* Now, **create a new test account** or select an existing one.
+* Click on the **settings** icon on top-right corner, and select integrations.
+* Create a private app (or open an existing one).
+* Navigate to **Scopes**.
+* Allow necessary permissions for **CMS, CRM, and Settings**.
+* Navigate to the auth page of your app settings.
+* Here you'll be able to find the app's **client ID** and [**access token**](https://developers.hubspot.com/docs/api/oauth/tokens).
+
+
+Access token will look something like this:
+
+```js
+pat-na1-217ee**-****-****-****-***
+```
+
+
+## Create Queries
+
+You can add queries to the newly created HubSpot datasource in one of the below ways:
+
+
+### Datasource Page
+You can add queries to Hubspot datasource by selecting the **New API +** button available on the datasource page.
+
+![](/img/hubspot-query1.png)
```

Comment:
Please rename the image. You could add some meaningful names like - hubspot-create-query-from-datasource-page.